backend/icpy/agent/helpers.py

```
@@ -305,16 +442,82 @@ def stream_chat_with_tools(self, messages: List[Dict[str, Any]],
+                    logger.warning(f"OpenAIStreamingHandler: Dropping empty user message at index {idx}")
+                    continue
+                conv.append({**m, "role": role, "content": content})
+            if dropped_users:
+                logger.info(f"OpenAIStreamingHandler: Dropped {dropped_users} empty user message(s) before request")
+
+            # Log a brief preview at INFO for easier troubleshooting
+            try:
+                preview = "\n".join([f"{i}: {m.get('role')} len={len(m.get('content','') or '')}" for i, m in enumerate(conv)])
+                logger.info("OpenAIStreamingHandler: Outbound messages preview\n" + preview)
+            except Exception:
+                pass
```

Ruff (0.13.1)

`try`-`except`-`pass` detected, consider logging the exception

(S110)